File: src/main/java/com/indico/mutation/GenerateSubmissionResult.java
package com.indico.mutation;

import com.apollographql.apollo.ApolloCall;
import com.apollographql.apollo.api.Response;
import com.indico.Async;
import com.indico.CreateSubmissionResultsGraphQLMutation;
import com.indico.IndicoClient;
import com.indico.Mutation;
import com.indico.entity.Submission;
import com.indico.jobs.Job;

import java.util.concurrent.CompletionException;
import java.util.concurrent.ExecutionException;

public class GenerateSubmissionResult implements Mutation<Job> {
    private final IndicoClient client;
    private int submissionId;
    private boolean retrieved;

    public GenerateSubmissionResult(IndicoClient client) {
        this.client = client;
    }

    public GenerateSubmissionResult submission(int submissionId) {
        this.submissionId = submissionId;
        return this;
    }

    public GenerateSubmissionResult submission(Submission submission) {
        this.submissionId = submission.id;
        return this;
    }

    @Override
    public Job execute() {
        try{
        ApolloCall<CreateSubmissionResultsGraphQLMutation.Data> apolloCall = this.client.apolloClient.mutate(CreateSubmissionResultsGraphQLMutation.builder()
                .submissionId(this.submissionId)
                .build());

        Response<CreateSubmissionResultsGraphQLMutation.Data> response = Async.executeSync(apolloCall).get();
        CreateSubmissionResultsGraphQLMutation.SubmissionResults submissionResults = response.data().submissionResults();
        String jobId = submissionResults.jobId();
        return new Job(this.client, jobId);  }
        catch (CompletionException | ExecutionException | InterruptedException ex){
            throw new RuntimeException("Call to generate the submission result failed", ex);
        }
    }
}

File: src/main/java/com/indico/mutation/GenerateSubmissionResult.kt
package com.indico.mutation

import com.indico.IndicoClient
import com.indico.entity.Submission
import com.indico.exceptions.IndicoMutationException
import com.indico.graphql.CreateSubmissionResultsGraphQL
import com.indico.query.Job

class GenerateSubmissionResult(private val client: IndicoClient) : Mutation<Job?, CreateSubmissionResultsGraphQL.Result>() {
    private var submissionId = 0
    fun submission(submissionId: Int): GenerateSubmissionResult {
        this.submissionId = submissionId
        return this
    }

    fun submission(submission: Submission): GenerateSubmissionResult {
        submissionId = submission.id
        return this
    }

    override fun execute(): Job? {
        return try {
            val call = CreateSubmissionResultsGraphQL(variables =
            CreateSubmissionResultsGraphQL.Variables(
                submissionId = this.submissionId
            ))

            val response = this.client.execute(call)
            handleErrors(response)
            val submissionResults = response.data?.submissionResults?:
                throw IndicoMutationException("Failed to generate submission results, no submission results returned.")
            val jobId: String = submissionResults.jobId!!
            Job(client, id = jobId, errors = emptyList())
         } catch (ex: RuntimeException) {
            throw IndicoMutationException("Call to generate the submission result failed", ex)
        }
    }

}